coderd/database/queries/notificationsinbox.sql

```diff
@@ -0,0 +1,71 @@
+-- name: GetUnreadInboxNotificationsByUserID :many
+--
+SELECT * FROM inbox_notifications WHERE
+	user_id = @user_id AND
+	read_at IS NULL AND
+	(@id_opt::UUID = '00000000-0000-0000-0000-000000000000'::UUID OR id > @id_opt::UUID)
+	ORDER BY created_at DESC
+	LIMIT (COALESCE(NULLIF(@limit_opt :: INT, 0), 25));
+
+-- name: GetInboxNotificationsByUserID :many
+SELECT * FROM inbox_notifications WHERE
+	user_id = @user_id AND
+	(@id_opt::UUID = '00000000-0000-0000-0000-000000000000'::UUID OR id > @id_opt::UUID)
+	ORDER BY created_at DESC
+	LIMIT (COALESCE(NULLIF(@limit_opt :: INT, 0), 25));
+
+-- name: GetInboxNotificationsByUserIDFilteredByTemplatesAndTargets :many
+-- Fetches inbox notifications for a user filtered by templates and targets
+-- param user_id: The user ID
+-- param templates: The template IDs to filter by - the template_id = ANY(@templates::UUID[]) condition checks if the template_id is in the @templates array
+-- param targets: The target IDs to filter by - the targets @> COALESCE(@targets, ARRAY[]::UUID[]) condition checks if the targets array (from the DB) contains all the elements in the @targets array
+SELECT * FROM inbox_notifications WHERE
+	user_id = @user_id AND
+	template_id = ANY(@templates::UUID[]) AND
+	targets @> COALESCE(@targets, ARRAY[]::UUID[]) AND
+	(@id_opt::UUID = '00000000-0000-0000-0000-000000000000'::UUID OR id > @id_opt::UUID)
+	ORDER BY created_at DESC
+	LIMIT (COALESCE(NULLIF(@limit_opt :: INT, 0), 25));
+
+-- name: GetUnreadInboxNotificationsByUserIDFilteredByTemplatesAndTargets :many
+-- param user_id: The user ID
+-- param templates: The template IDs to filter by - the template_id = ANY(@templates::UUID[]) condition checks if the template_id is in the @templates array
+-- param targets: The target IDs to filter by - the targets @> COALESCE(@targets, ARRAY[]::UUID[]) condition checks if the targets array (from the DB) contains all the elements in the @targets array
+SELECT * FROM inbox_notifications WHERE
+	user_id = @user_id AND
+	template_id = ANY(@templates::UUID[]) AND
+	targets @> COALESCE(@targets, ARRAY[]::UUID[]) AND
+	read_at IS NULL AND
+	(@id_opt::UUID = '00000000-0000-0000-0000-000000000000'::UUID OR id > @id_opt::UUID)
+	ORDER BY created_at DESC
+	LIMIT (COALESCE(NULLIF(@limit_opt :: INT, 0), 25));
+
+-- name: GetInboxNotificationByID :one
+SELECT * FROM inbox_notifications WHERE id = $1;
+
+-- name: CountUnreadInboxNotificationsByUserID :one
+SELECT COUNT(*) FROM inbox_notifications WHERE user_id = $1 AND read_at IS NULL;
```

Comment:
Where do we plan to use this query? If we're just displaying the unread notifications count in the UI, would it suffice to just call `len(GetUnreadInboxNotificationsByUserID()`? My assumption is that the UI will be fetching unread notifications anyway.

Reply:
This query will be called in two cases : 

- An endpoint to get the number of unread notifications (without any notifications - just the count)
- Whenever you want to list the number of notifications (unread or not, filtered or not) - we'll include the `unread_count`